### PR TITLE
Remove legacy tests and move to new framework

### PR DESCRIPTION
**Pull request by shariarriday**
Move some legacy tests to new framework

- string_algorithms
- string_view
- type_id_list
- typed_behavior
- typed_message_view
- unit
- uri
- uuid

### CODE REVIEW

File: libcaf_core/caf/string_view.test.cpp [hunk 1]
```diff
@@ -0,0 +1,211 @@
+// This file is part of CAF, the C++ Actor Framework. See the file LICENSE in
+// the main distribution directory for license terms and copyright or visit
+// https://github.com/actor-framework/actor-framework/blob/master/LICENSE.
+
+#include "caf/string_view.hpp"
+
+#include "caf/test/caf_test_main.hpp"
```
Comment (Neverlord) [inline libcaf_core/caf/string_view.test.cpp:7]:
```suggestion
```

File: libcaf_core/caf/string_algorithms.test.cpp [hunk 2]
```diff
@@ -0,0 +1,83 @@
+// This file is part of CAF, the C++ Actor Framework. See the file LICENSE in
+// the main distribution directory for license terms and copyright or visit
+// https://github.com/actor-framework/actor-framework/blob/master/LICENSE.
+
+#include "caf/string_algorithms.hpp"
+
+#include "caf/test/caf_test_main.hpp"
```
Comment (Neverlord) [inline libcaf_core/caf/string_algorithms.test.cpp:7]:
```suggestion
```

Is this added automatically? In that case, please drop from the script.

File: libcaf_core/caf/string_algorithms.test.cpp [hunk 3]
```diff
@@ -0,0 +1,83 @@
...
+  caf::split(result, str, ",", token_compress_on);
+  return result;
+}
+
+std::string join(str_list vec) {
+  return caf::join(vec, ",");
+}
+
+TEST("splitting") {
+  check_eq(split(""), str_list({""}));
```
Comment (Neverlord) [inline libcaf_core/caf/string_algorithms.test.cpp:36]:
```
  check_eq(split(""), str_list{""});
```

We've used the extra parens to protect against the CHECK_EQ macro before, so they're no longer needed. Same in the test below

File: libcaf_core/caf/type_id_list.test.cpp [hunk 4]
```diff
@@ -2,11 +2,12 @@
 // the main distribution directory for license terms and copyright or visit
 // https://github.com/actor-framework/actor-framework/blob/master/LICENSE.
 
-#define CAF_SUITE type_id_list
-
 #include "caf/type_id_list.hpp"
 
-#include "core-test.hpp"
+#include "caf/test/caf_test_main.hpp"
```
Comment (Neverlord) [inline libcaf_core/caf/type_id_list.test.cpp:7]:
```suggestion
```

File: libcaf_core/caf/typed_behavior.test.cpp [hunk 5]
```diff
@@ -2,24 +2,27 @@
 // the main distribution directory for license terms and copyright or visit
 // https://github.com/actor-framework/actor-framework/blob/master/LICENSE.
 
-#define CAF_SUITE typed_behavior
-
 #include "caf/typed_behavior.hpp"
 
-#include "caf/typed_actor.hpp"
+#include "caf/test/caf_test_main.hpp"
```
Comment (Neverlord) [inline libcaf_core/caf/typed_behavior.test.cpp:7]:
```suggestion
```

File: libcaf_core/caf/typed_message_view.test.cpp [hunk 6]
```diff
@@ -2,41 +2,44 @@
 // the main distribution directory for license terms and copyright or visit
 // https://github.com/actor-framework/actor-framework/blob/master/LICENSE.
 
-#define CAF_SUITE typed_message_view
-
 #include "caf/typed_message_view.hpp"
 
-#include "caf/message.hpp"
+#include "caf/test/caf_test_main.hpp"
```
Comment (Neverlord) [inline libcaf_core/caf/typed_message_view.test.cpp:7]:
```suggestion
```

File: libcaf_core/caf/unit.test.cpp [hunk 7]
```diff
@@ -2,19 +2,21 @@
 // the main distribution directory for license terms and copyright or visit
 // https://github.com/actor-framework/actor-framework/blob/master/LICENSE.
 
-#define CAF_SUITE unit
-
 #include "caf/unit.hpp"
 
+#include "caf/test/caf_test_main.hpp"
```
Comment (Neverlord) [inline libcaf_core/caf/unit.test.cpp:7]:
```suggestion
```

File: libcaf_core/caf/uri.test.cpp [hunk 8]
```diff
@@ -2,16 +2,15 @@
 // the main distribution directory for license terms and copyright or visit
 // https://github.com/actor-framework/actor-framework/blob/master/LICENSE.
 
-#define CAF_SUITE uri
-
 #include "caf/uri.hpp"
 
+#include "caf/test/caf_test_main.hpp"
```
Comment (Neverlord) [inline libcaf_core/caf/uri.test.cpp:7]:
```suggestion
```

File: libcaf_core/caf/uri.test.cpp [hunk 9]
```diff
@@ -196,44 +197,44 @@ uri operator"" _u(const char* cstr, size_t cstr_len) {
   std::string_view str{cstr, cstr_len};
   auto err = parse(str, result);
   if (err)
-    CAF_FAIL("error while parsing " << str << ": " << to_string(err));
+    test::runnable::current().fail("error while parsing {} : {}", str,
```
Comment (Neverlord) [inline libcaf_core/caf/uri.test.cpp:200]:
```suggestion
    test::runnable::current().fail("error while parsing {}: {}", str,
```

File: libcaf_core/caf/uuid.test.cpp [hunk 10]
```diff
@@ -0,0 +1,180 @@
...
+#include "caf/json_reader.hpp"
+#include "caf/json_writer.hpp"
+
+using namespace caf;
+
+namespace {
+
+uuid operator"" _uuid(const char* cstr, size_t cstr_len) {
+  if (cstr_len != 36)
+    test::runnable::current().fail("malformed test input");
```
Comment (Neverlord) [inline libcaf_core/caf/uuid.test.cpp:20]:
```
  auto& this_test = test::runnable::current();
  if (cstr_len != 36)
    this_test.fail("malformed test input");
```

(and use `this_test` below as well)

**Review by Neverlord — CHANGES_REQUESTED** [review]

File: libcaf_core/caf/uuid.test.cpp (hunk 10 above)
Comment (shariarriday) [reply libcaf_core/caf/uuid.test.cpp]: I have refactored the tests a little bit for this change. Previously, the uuid's were being generated from the fixture and the  `runnable` object was not available at that point which causes the tests to fail. Moving the object declarations inside the test solves this issue. Please let me know if you have any feedback regarding this refactor.

Comment (Neverlord) [reply libcaf_core/caf/uuid.test.cpp]: Good call, LGTM.

**Review by Neverlord — APPROVED** [review]

**Pull request merged**